testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvSinglenodeDeltaLakeOss.java

```diff
@@ -0,0 +1,137 @@
+
+    private final DockerFiles dockerFiles;
+    private final PortBinder portBinder;
+    private final String hadoopImagesVersion;
+    private final DockerFiles.ResourceProvider configDir;
+
+    @Inject
+    public EnvSinglenodeDeltaLakeOss(
+            Standard standard,
+            Hadoop hadoop,
+            DockerFiles dockerFiles,
+            EnvironmentConfig config,
+            PortBinder portBinder,
+            Minio minio)
+    {
+        super(ImmutableList.of(standard, hadoop, minio));
+        this.dockerFiles = requireNonNull(dockerFiles, "dockerFiles is null");
+        this.portBinder = requireNonNull(portBinder, "portBinder is null");
+        this.hadoopImagesVersion = requireNonNull(config, "config is null").getHadoopImagesVersion();
+        this.configDir = dockerFiles.getDockerFilesHostDirectory("conf/environment/singlenode-delta-lake-oss");
+    }
+
+    @Override
+    public void extendEnvironment(Environment.Builder builder)
+    {
+        // Using hdp3.1 so we are using Hive metastore with version close to versions of  hive-*.jars Spark uses
+        builder.configureContainer(HADOOP, container -> {
+            container.setDockerImageName("ghcr.io/trinodb/testing/hdp3.1-hive:" + hadoopImagesVersion);
+        });
+
+        builder.configureContainer(COORDINATOR, container -> container
+                .withCopyFileToContainer(forHostPath(configDir.getPath("hive.properties")), CONTAINER_PRESTO_ETC + "/catalog/hive.properties")
+                .withCopyFileToContainer(forHostPath(configDir.getPath("delta.properties")), CONTAINER_PRESTO_ETC + "/catalog/delta.properties"));
+
+        builder.configureContainer(TESTS, dockerContainer -> {
+            dockerContainer.withEnv("S3_BUCKET", S3_BUCKET_NAME)
+                    .withCopyFileToContainer(
+                            forHostPath(dockerFiles.getDockerFilesHostPath("conf/tempto/tempto-configuration-for-hive3.yaml")),
+                            CONTAINER_TEMPTO_PROFILE_CONFIG);
+        });
+
+        builder.addContainer(createSparkContainer())
+                // Ensure Hive metastore is up; Spark needs to access it during startup
+                .containerDependsOn(SPARK_CONTAINER_NAME, HADOOP);
+
+        // Initialize buckets in Minio
+        FileAttribute<Set<PosixFilePermission>> posixFilePermissions = PosixFilePermissions.asFileAttribute(PosixFilePermissions.fromString("rw-r--r--"));
+        Path minioBucketDirectory;
+        try {
+            minioBucketDirectory = Files.createTempDirectory("presto-ci-test", posixFilePermissions);
+            minioBucketDirectory.toFile().deleteOnExit();
+        }
+        catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+        builder.configureContainer(MINIO_CONTAINER_NAME, container ->
+                container.withCopyFileToContainer(forHostPath(minioBucketDirectory), "/data/" + S3_BUCKET_NAME));
```

Comment:
This should go in the `commin/Minio` file. It's not clear here what's special about the `/data/` directory

Reply:
isn't `/data` where MinIO docker container stores buckets?

Reply:
`/data` is the argument given to minio server to point the directory where the buckets are physically stored.

The Docker container process is started with:
 
```
                .withCommand("server", "--address", format("0.0.0.0:%d", MINIO_PORT), "/data")
```

I think that this customisation is specific for `EnvSinglenodeDeltaLakeOss` and not a common setting for `MinioContainer`.

Reply:
We should at least have `/data/` stored in a static class variable on `Minio` then